### PR TITLE
Adds documentation for script-only mode and templates

### PR DESCRIPTION
**Pull request by nick-l-o3de**
## Change summary

* The welcome guide now links to the new page about templates, which is a list of which templates are available and what they do.
* The new templates page includes instructions on how to use non-project templates.
* The new script-only 'Quick Start' page explains what a script-only project is, how to create one, and how to use them, as well as includes technical details about how the system works.

### Submission Checklist:

* [X] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [X] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [X] **Consistency** - Does the content consistently follow the [Style Guide](https://o3de.org/docs/contributing/to-docs/style-guide/quick-reference)?
* [X] **Help the user** - Does the documentation show the user something *meaningful*?

---
feel free to supply feedback or edit the changes.

### CODE REVIEW

**Comment by nick-l-o3de** [comment]
If you'd like to see the changes in preview, you can see the new pages in the following location

1. Linked to by the word 'templates' in the Getting Started /Creating a project using project manager page, in the following Section
> Under Select a Project Template, you can choose a project template with a pre-configured selection of Gems. You can modify the list of Gems enabled for your project by choosing the Configure Gems button. This tutorial uses the default template and its pre-configured selection of common Gems. Other templates are available. For more information about templates, and which templates are available, refer to the documentation on [Templates](https://deploy-preview-2536--o3deorg.netlify.app/docs/user-guide/build/templates/) in the User Guide.

2.  In the user guide, under the 'build' topic, there are 2 new pages - Templates, and Script-Only mode

File: content/docs/user-guide/build/templates.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+### ScriptOnlyProject
+This has an identical gem selection to DefaultProject, but disables all C++ compiling and linking.  The project is still able to use the Lua and Script Canvas capabilities of the engine, but cannot have custom c++ code embedded or use other gems or modules that are not pre-compiled.  It can be used to get a quick look at the engine and editor without having to compile a project, download 3rd party libraries, or install a compiler or linker.   You can convert a Script Only project to a regular project at a later time. 
+
+See the [Script-only 'Quick Start' Projects](/docs/user-guide/build/script-only-projects.md) documentation for more information about this feature.
+
+### MinimalProject
+This has as few gems active as possible, and is a bare-bones starting point.  It is not recommended to use this template as-is, but rather start with it and then activate what gems you might need for your project.   Because it activates as little as possible, and includes as few other gems as possible, it has the quickest startup time and asset compilation time.
+
+## Notable gem templates included with the engine
```
Comment (AMZN-alexpete) [inline content/docs/user-guide/build/templates.md:28]:
```suggestion
## Notable Gem templates included with the engine
```
I believe we usually capitalize the word Gem in headers.

**Review by AMZN-alexpete — APPROVED** [review]

File: content/docs/user-guide/build/templates.md (hunk 1 above)
Comment (nick-l-o3de) [reply content/docs/user-guide/build/templates.md:28]: Will fix!

**Comment by nick-l-o3de** [comment]
Looks like this is ready to merge?

**Pull request merged**